rust-checks/src/analysis.rs

```diff
@@ -23,6 +27,35 @@ pub fn analyze_nix_files(
     Ok(serde_json::to_string(&report)?)
 }
 
+/// Runs given analyzer on log file for each attr, if exists
+pub fn analyze_log_files(
+    attrs: Vec<Attr>,
+    analyzer: LogFileAnalyzer,
+) -> Result<String, Box<dyn Error>> {
+    let mut report: HashMap<String, Report> = HashMap::new();
+
+    for attr in attrs
+        .iter()
+        .filter(|a| a.output.is_some())
+        .filter(|a| Path::new(&a.output.as_ref().unwrap()).exists())
+    {
+        let mut program = Command::new("nix")
+            .args(&[
+                "--experimental-features",
+                "nix-command",
+                "log",
+                &attr.output.as_ref().unwrap(),
+            ])
+            .stdout(Stdio::piped())
+            .spawn()?;
+
+        let out = BufReader::new(program.stdout.take().unwrap());
```

Comment:
```
RUST_BACKTRACE=1 nix run -f /home/mcgibbon/projects/nixpkgs-hammering -c nixpkgs-hammer -f . --json  python38Packages.scipy
warning: --- Large path ---------------------------------------------------------------------------------------------------------------------- nix
dumping very large path (> 256 MiB); this may run out of memory
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: InvalidData, error: "stream did not contain valid UTF-8" }', src/bin/stale-substitute.rs:23:27
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: core::ops::function::impls::<impl core::ops::function::FnMut<A> for &mut F>::call_mut
   4: <alloc::vec::Vec<T> as alloc::vec::SpecFromIter<T,I>>::from_iter
   5: stale_substitute::analyze_single_file
   6: nixpkgs_hammering_ast_checks::analysis::analyze_log_files
   7: stale_substitute::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Rule 'stale-substitute'' failed with input '[{"name": "python38Packages.scipy", "location": {"file": "/home/mcgibbon/projects/nixpkgs/pkgs/development/python-modules/scipy/default.nix", "line": 57, "column": null}, "drv": "/nix/store/8a8rngphi3dj2yxd2l4vi864kv3znw60-python3.8-scipy-1.6.0.drv", "output": "/nix/store/6cjl7khra87587r0jvnx3gsf338ps7b2-python3.8-scipy-1.6.0"}]'
Traceback (most recent call last):
  File "/nix/store/nb5qchz440gcgaw4zdn8lzhm55m36d7c-nixpkgs-hammer/bin/.nixpkgs-hammer-wrapped", line 416, in <module>
    main(args)
  File "/nix/store/nb5qchz440gcgaw4zdn8lzhm55m36d7c-nixpkgs-hammer/bin/.nixpkgs-hammer-wrapped", line 345, in main
    external_reports = run_external_checks(
  File "/nix/store/nb5qchz440gcgaw4zdn8lzhm55m36d7c-nixpkgs-hammer/bin/.nixpkgs-hammer-wrapped", line 121, in run_external_checks
    json_text = subprocess.check_output([rule], text=True, input=encoded_attrs)
  File "/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.7/lib/python3.8/subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.7/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['stale-substitute']' returned non-zero exit status 101.
```

Something here appears to not handle non-UTF8 text correctly.

Reply:
Fixed

Reply:
Could you clarify what the issue was? It is not obvious to me from the changes.